storage-server-api/src/main/java/org/jetbrains/ruby/codeInsight/types/storage/server/DatabaseProvider.kt

```diff
@@ -0,0 +1,57 @@
+package org.jetbrains.ruby.codeInsight.types.storage.server
+
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.transactions.TransactionManager
+import java.sql.Connection
+
+object DatabaseProvider {
+    private val IN_MEMORY_URL = "jdbc:h2:mem:test"
+    private val IN_MEMORY_DRIVER = "org.h2.Driver"
+    private val MYSQL_URL = System.getProperty("mysql.url",
+            "jdbc:mysql://localhost:3306/" +
+                    "ruby_type_contracts" +
+                    "?serverTimezone=UTC&nullNamePatternMatchesAll=true&useSSL=false")
+    private val MYSQL_DRIVER = "com.mysql.cj.jdbc.Driver"
+    private val MYSQL_USER = System.getProperty("mysql.user.name", "rubymine")
+    private val MYSQL_PASSWORD = System.getProperty("mysql.user.password", "rubymine")
+
+
+    fun connect(inMemory: Boolean = false,
+                setupConnection: ((Connection) -> Unit)? = null,
```

Comment:
I am not sure you need these params. In java classes they were specified just because kotlin did not generate enough overloads for this method. I'd remove all params (except `inMemory`) and cleanup the code inside

Reply:
But what about `CallStatCompletionTest`? It passes additional parameters to `DatabaseProvider`.

Reply:
>  In java classes they were specified just because kotlin did not generate enough overloads for this method

This is what I was talking about, in `CallStatCompletionTest` I've just re-initialised the default values.

Reply:
Thx for explanation, now I get it.